### ISSUE TITLE
IMauiContext not found when calling IImageSourceHandler.LoadImageAsync()

### ISSUE THREAD

**Issue by IgorKravchenko10** [dotnet/maui#7646]
### Description

Full code is [here](https://1drv.ms/u/s!Aun0lW86yXKJii5V0LpEscLsxEGP?e=MfpJXp).
I have a several custom views with `ImageSource` bindable properties. For example, `MaterialEntry`.
To use image in `MaterialEntryHandler.Android.cs` I need to convert ImageSource to android BitmapDrawable. A have an extension method in `TcuMaui.Platforms.Android.Extensions.Extensions` class called ToBitmap. 

```
public static IImageSourceHandler GetHandler(this ImageSource imageSource)
        {
            if (imageSource is FileImageSource)
                return new FileImageSourceHandler();
            if (imageSource is StreamImageSource)
                return new StreamImagesourceHandler();
            if (imageSource is UriImageSource)
                return new ImageLoaderSourceHandler();
            if (imageSource is FontImageSource)
                return new FontImageSourceHandler();
            return null;
        }

        public static Task<Bitmap> ToBitmap(this ImageSource imageSource, CancellationToken cancellationToken = default)
        {
            if (imageSource == null)
                return null;
            IImageSourceHandler handler = imageSource.GetHandler();
            //here I get IMauiContext not found
            return handler.LoadImageAsync(imageSource, MainActivity.Instance, cancellationToken);
        }

        public static async Task<BitmapDrawable> ToBitmapDrawable(this ImageSource imageSource, CancellationToken cancellationToken = default)
        {
            if (imageSource == null)
                return null;
            Bitmap bitmap = await imageSource.ToBitmap(cancellationToken);
            return new BitmapDrawable(MainActivity.Instance.Resources, bitmap);
        }
```
After calling `handler.LoadImageAsync(imageSource, MainActivity.Instance, cancellationToken)` I get an exception IMauiContext not found.
Is there any another method to get bitmap from ImageSource?
[This issue on Microsoft Q&A](https://docs.microsoft.com/en-us/answers/questions/865667/imauicontext-not-found-when-trying-to-get-bitmap-f.html)
Thanks.

### Steps to Reproduce

1. Create ImageSource bindable property.
2. Create IImageSourceHandler for ImageSource in platform code.
3. Call LoadImageAsync() method of IImageSourceHandler.

### Version with bug

6.0 Release Candidate 3

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

### Relevant log output

```shell
System.InvalidOperationException: IMauiContext not found.
 [mono-rt]    at Microsoft.Maui.Controls.ViewExtensions.RequireMauiContext(Element element, Boolean fallbackToAppMauiContext) in D:\a\_work\1\s\src\Controls\src\Core\ViewExtensions.cs:line 200
 [mono-rt]    at Microsoft.Maui.Controls.ViewExtensions.RequireFontManager(Element element, Boolean fallbackToAppMauiContext) in D:\a\_work\1\s\src\Controls\src\Core\ViewExtensions.cs:line 217
 [mono-rt]    at Microsoft.Maui.Controls.Compatibility.Platform.Android.FontImageSourceHandler.LoadImageAsync(ImageSource imagesource, Context context, CancellationToken cancelationToken) in D:\a\_work\1\s\src\Compatibility\Core\src\Android\Renderers\FontImageSourceHandler.cs:line 36
```

**Comment by kristinx0211** [comment]
confirm this error happen when when calling `IImageSourceHandler.LoadImageAsync()` with above project.

**Comment by the8thC** [comment]
Hello, we are developing [DevExpress .NET MAUI controls](https://www.devexpress.com/maui/) and experienced the same issue. This issue has a high priority for us and we do not have a workaround. Thank you for your help.

**Comment by IgorKravchenko10** [comment]
> Hello, we are developing [DevExpress .NET MAUI controls](https://www.devexpress.com/maui/) and experienced the same issue. This issue has a high priority for us and we do not have a workaround. Thank you for your help.

Check my latest answer [here](https://docs.microsoft.com/en-us/answers/questions/865667/imauicontext-not-found-when-trying-to-get-bitmap-f.html) from Jun 09 2022. There is a workaround.

###### Our company used your controls for many years. But now your country killing our people. Sad to know we paid to sponsors of killers.

**Comment by TimLandskron** [comment]
Currently, we are also facing this issue on the iOS platform.
The FontImageSourceHandler.LoadImageAsync() throws the same exception that the IMauiContext is not found.

Could it possibly be fixed or do we need to find a workaround?

**Comment by mattleibow** [comment]
The issue is with the fact that you have constructed an `ImageSource` object and then using _just_ that to try and render an image. This operation requires at least the font manager, and the only way to find it is to search up the hierarchy for a view and use that handler.

The handler that is being use in your code is the compatibility handler and the API did not allow for a maui context.

The new/correct way to load images is using the `ImageSourceService` and derived types. There is also a set of useful extension methods in the `ImageSourceExtensions` type.

TL;DR: the ImageSourceHandlers are compat-only. They need a maui context. They need to have a view parent that has a maui context. Image source services ae the correct way.
